### ISSUE TITLE
[VersionControl] Error pushing big files using Git

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7102]
Trying to commit and push big files (more than 100MB), sometimes we get the following error:

```
--- End of inner exception stack trace ---
at MonoDevelop.VersionControl.Git.GitRepository.RetryUntilSuccess (MonoDevelop.Core.ProgressMonitor monitor, System.Action`1[T] func, System.Action onRetry) [0x00166] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:657`
`at MonoDevelop.VersionControl.Git.GitRepository.Push (MonoDevelop.Core.ProgressMonitor monitor, System.String remote, System.String remoteBranch) [0x00068] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1210`
`at MonoDevelop.VersionControl.Git.GitService+<>c__DisplayClass3_1.<Push>b__0 (System.Object <p0>) [0x00002] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitService.cs:66`
`ERROR [2019-02-08 09:25:42Z]: unpacking the sent packfile failed on the remote`
`MonoDevelop.VersionControl.VersionControlException: unpacking the sent packfile failed on the remote ---> LibGit2Sharp.LibGit2SharpException: unpacking the sent packfile failed on the remote`
`at LibGit2Sharp.Core.Ensure.HandleError (System.Int32 result) [0x00064] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Ensure.cs:155`
`at LibGit2Sharp.Core.Ensure.ZeroResult (System.Int32 result) [0x0000c] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Ensure.cs:173`
`at LibGit2Sharp.Core.Proxy.git_remote_push (LibGit2Sharp.Core.Handles.RemoteHandle remote, System.Collections.Generic.IEnumerable`1[T] refSpecs, LibGit2Sharp.Core.GitPushOptions opts) [0x0002a] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Proxy.cs:2240
at LibGit2Sharp.Network.Push (LibGit2Sharp.Remote remote, System.Collections.Generic.IEnumerable`1[T] pushRefSpecs, LibGit2Sharp.PushOptions pushOptions) [0x00065] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Network.cs:372`
`at LibGit2Sharp.Network.Push (LibGit2Sharp.Remote remote, System.String pushRefSpec, LibGit2Sharp.PushOptions pushOptions) [0x0000d] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Network.cs:331`
`at MonoDevelop.VersionControl.Git.GitRepository+<>c__DisplayClass83_0.<Push>b__1 (MonoDevelop.VersionControl.Git.GitCredentialsType credType) [0x0000d] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1211`
`at MonoDevelop.VersionControl.Git.GitRepository.RetryUntilSuccess (MonoDevelop.Core.ProgressMonitor monitor, System.Action`1[T] func, System.Action onRetry) [0x00017] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:621
--- End of inner exception stack trace ---
at MonoDevelop.VersionControl.Git.GitRepository.RetryUntilSuccess (MonoDevelop.Core.ProgressMonitor monitor, System.Action`1[T] func, System.Action onRetry) [0x00166] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:657`
`at MonoDevelop.VersionControl.Git.GitRepository.Push (MonoDevelop.Core.ProgressMonitor monitor, System.String remote, System.String remoteBranch) [0x00068] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1210`
`at MonoDevelop.VersionControl.Git.GitService+<>c__DisplayClass3_1.<Push>b__0 (System.Object <p0>) [0x00002] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitService.cs:66`
`ERROR [2019-02-08 09:25:42Z]: unpacking the sent packfile failed on the remote`
`MonoDevelop.VersionControl.VersionControlException: unpacking the sent packfile failed on the remote ---> LibGit2Sharp.LibGit2SharpException: unpacking the sent packfile failed on the remote`
`at LibGit2Sharp.Core.Ensure.HandleError (System.Int32 result) [0x00064] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Ensure.cs:155`
`at LibGit2Sharp.Core.Ensure.ZeroResult (System.Int32 result) [0x0000c] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Ensure.cs:173`
`at LibGit2Sharp.Core.Proxy.git_remote_push (LibGit2Sharp.Core.Handles.RemoteHandle remote, System.Collections.Generic.IEnumerable`1[T] refSpecs, LibGit2Sharp.Core.GitPushOptions opts) [0x0002a] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Proxy.cs:2240
at LibGit2Sharp.Network.Push (LibGit2Sharp.Remote remote, System.Collections.Generic.IEnumerable`1[T] pushRefSpecs, LibGit2Sharp.PushOptions pushOptions) [0x00065] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Network.cs:372`
`at LibGit2Sharp.Network.Push (LibGit2Sharp.Remote remote, System.String pushRefSpec, LibGit2Sharp.PushOptions pushOptions) [0x0000d] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Network.cs:331`
`at MonoDevelop.VersionControl.Git.GitRepository+<>c__DisplayClass83_0.<Push>b__1 (MonoDevelop.VersionControl.Git.GitCredentialsType credType) [0x0000d] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1211`
`at MonoDevelop.VersionControl.Git.GitRepository.RetryUntilSuccess (MonoDevelop.Core.ProgressMonitor monitor, System.Action`1[T] func, System.Action onRetry) [0x00017] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:621
--- End of inner exception stack trace ---
at MonoDevelop.VersionControl.Git.GitRepository.RetryUntilSuccess (MonoDevelop.Core.ProgressMonitor monitor, System.Action`1[T] func, System.Action onRetry) [0x00166] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:657
at MonoDevelop.VersionControl.Git.GitRepository.Push (MonoDevelop.Core.ProgressMonitor monitor, System.String remote, System.String remoteBranch) [0x00068] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1210
at MonoDevelop.VersionControl.Git.GitService+<>c__DisplayClass3_1.b__0 (System.Object ) [0x00002] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitService.cs:66
```

The exception is **LibGit2SharpException: unpacking the sent packfile failed on the remote**.

Regression #7078

> VS bug [#789299](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789299)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1594 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)
